src/plugins/discover/public/components/log_explorer_tabs/log_explorer_tabs.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiTab, EuiTabs, useEuiTheme } from '@elastic/eui';
+import type { DataViewSpec } from '@kbn/data-views-plugin/public';
+import { AllDatasetsLocatorParams, ALL_DATASETS_LOCATOR_ID } from '@kbn/deeplinks-observability';
+import { i18n } from '@kbn/i18n';
+import React, { MouseEvent } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { map } from 'rxjs';
+import { DiscoverAppLocatorParams, DISCOVER_APP_LOCATOR } from '../../../common';
+import type { DiscoverServices } from '../../build_services';
+
+export interface LogExplorerTabsParams {
+  columns?: string[];
+  sort?: string[][];
+  dataViewSpec?: DataViewSpec;
+}
+
+export interface LogExplorerTabsProps {
+  services: Pick<DiscoverServices, 'share' | 'data'>;
+  params: LogExplorerTabsParams;
+  selectedTab: 'discover' | 'log-explorer';
+}
+
+export const LogExplorerTabs = ({ services, params, selectedTab }: LogExplorerTabsProps) => {
```

Comment:
I'd prefer to keep this specific to Logs Explorer since it's a temporary measure which we intend on removing sooner than later, and we don't want others to begin relying on the functionality if we make it generic. It's the same reason I kept it outside of the customization framework and went with hardcoding instead.